src/validation.cpp

```diff
@@ -2304,13 +2304,19 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
                 [&cache] (CScript const & owner, DCT_ID tokenID) {
                     return cache.GetBalance(owner, tokenID);
                 },
-                [&cache, &block] (CScript const & to, DCT_ID poolID, uint8_t type, CTokenAmount amount) {
+                [&cache, &block] (CScript const & to, CScript const & from, DCT_ID poolID, uint8_t type, CTokenAmount amount) {
+                    if (from != CScript()) {
+                        auto res = cache.SubBalance(from, amount);
```

Comment:
`SubBalance` will erase zeroes balance, is there needed to check them in `ApplyUpdatePoolPairTx`

Reply:
The `ApplyUpdatePoolPairTx` check that we do not create a `0@token` reward as `WriteBy<Reward>` in the `CPoolPairView`. `SubBalance` will subtract from the pool owner address and will not impact the `1@token` record.

Reply:
Got'cha. So you can remove it before wirte? I mean we don't need to keep zeroes.

Reply:
Correct. So we do not write zero amount rewards.